### ISSUE TITLE
Inconsistent argument update -s/--sibling

### ISSUE THREAD

**Issue by mih** [datalad/datalad#3832]
Other commands (`siblings`, `create-sibling-gitlab`) use `-s/--name`.

**Comment by yarikoptic** [comment]
Would be nice to unify. 

There is also related overlap in `--to`: `--to=REVISION` being used by `diff`, while `publish` and now `push` use `--to=sibling` .  I have not researched yet but I would not be surprised if we would like to get `--to=REVISION` for publish/push to publish specific ranges of changes instead of the overall state.

I somewhat dislike having `-s` but `--name`, and even though tautological for those commands having `--sibling` would be consistent with other commands. Otherwise I need to remember to use short version (`-s`) to not use an incorrect long one.

**Comment by yarikoptic** [comment]
FWIW, I grew to like (actually I think the API grew to be that as well) conda's consistent use of `-n` to describe name of the environment.  Besides `conda activate NAME` all other commands IIRC ([chatsheet](https://docs.conda.io/projects/conda/en/4.6.0/_downloads/52a95608c49671267e40c689e0bc00ca/conda-cheatsheet.pdf)) have it as `-n|--name NAME` so I do not need to think twice what it could be (`-e vs -n`) or positional or an option. I feel that ideally we should unify similarly, that it is always `-s NAME` in some cases even if it iseems to be a good idea to make it a positional argument.  In tricky cases (I am not sure if we filed an issue) where we might eventually want something like `push  --from X --to Y` it could still be `push --from|--from-sibling X -s Y`.

**Comment by jsheunis** [comment]
@yarikoptic @mih I'm looking into this. So do we want to purely change the argument for `update` to `-s/--name` ([here](https://github.com/datalad/datalad/blob/b3acf7ca3dadeacc15aa6308d4d51ff7a973011b/datalad/distribution/update.py#L131)) or do we want to reconsider the whole `-s/--name` vs `-n/--name`. In the latter case, should that be a separate issue? In the former, should I make the PR against the `maint` branch?

**Comment by yarikoptic** [comment]
no options rename should be for `maint` .  While at it we might want to go for uniform `-s/--sibling-name` ?

**Comment by jsheunis** [comment]
Is that a question or a suggestion to implement? I don't have a specific preference and would defer to datalad elders on this :)

I've done searches for `--name` and `--sibling` and looks like changes will need to be made in these modules:
- `update`
- `siblings`
- `create_sibling`
- `create_sibling_github`
- `create_sibling_gitlab`
- `create_sibling_ria`

Am I missing something?

**Comment by yarikoptic** [comment]
> Is that a question or a suggestion to implement?

well -- both, since ideally we should boil it down to something all of us "accept" to be implemented.

> Am I missing something?

I guess there could be more in some extensions, eg datalad-osf?  but that would/could be later to find/unify

**Comment by adswa** [comment]
Re-reading this, it sounds like there was consensus or at least no one voiced any argument against @yarikoptic's suggestion of ``-s/--sibling-name`` across commands?
Anyone against this please downvote (:-1:) this comment, anyone indifferent react with :eyes:, and anyone for it react with :+1:

**Comment by mih** [comment]
I am in favor of having `-s/--sibling-name` to be available across commands uniformly. In terms of implementing this, I think we should keep the respective non-uniform long-form for now, and amend the respective docstrings to label them as deprecated.

It would be good to start a design document on "standard arguments" and have this be the first one listed.

**Comment by adswa** [comment]
I will start a design document

**Comment by yarikoptic** [comment]
FWIW note that [handbook's cheatsheet](http://handbook.datalad.org/en/latest/basics/101-136-cheatsheet.html#cheat) already lists some of the common/consistent options to start with.

**Comment by adswa** [comment]
Yes, and by the looks of it, it is in need of an update :roll_eyes:

**Comment by adswa** [comment]
what are you voting for, @yarikoptic ?
